### PR TITLE
Update end to end tests

### PR DESCRIPTION
**Pull request by f2prateek**
Use the testing harness that uses the new schema webhook.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,9 +9,14 @@ dependencies:
 	bundle install --verbose
 
 # Run all tests and checks (including linters).
-check:
+check: install  # Installation required for testing binary
 	bundle exec rake
+	sh .buildscript/e2e.sh
```
Comment (f2prateek) [inline Makefile]: It may be to nice run this e2e test only on CI, since it'll fail locally if you're using a non-Linux machine.

Comment (rohitpaulk) [reply Makefile]: Is it okay to rely on the `RUN_E2E_TESTS` env var not being set? The script will terminate early if so.

**Comment by rohitpaulk** [comment]
One note - now that we've removed tests with the e2e tag, we should remove [this line](https://github.com/segmentio/analytics-ruby/blob/244e1adf40e1936c7bc475c17400fc2b9e619009/Rakefile#L7) too

**Comment by f2prateek** [comment]
@rohitpaulk I removed this line as you suggested https://github.com/segmentio/analytics-ruby/blob/244e1adf40e1936c7bc475c17400fc2b9e619009/Rakefile#L7. I also updated the tester to use env vars instead of cli args. Looks like I may have broken something - mind taking another look when you get the chance?

**Comment by f2prateek** [comment]
weird, went back to flags and it looks like something isn't quite working still :(

**Comment by rohitpaulk** [comment]
Hmm, weird indeed... Will take a look today/tomorrow.

**Comment by f2prateek** [comment]
thanks!

**Comment by f2prateek** [comment]
I was able to fix the issue!

**Pull request merged**